src/stores/contextKeyStore.ts

```diff
@@ -0,0 +1,92 @@
+import { defineStore } from 'pinia'
+
+/**
+ * Store for context keys.
+ * Holds arbitrary key-value pairs globally throughout the app.
+ */
+export const useContextKeyStore = defineStore('contextKeys', {
```

Comment:
Please use composition API to write pinia store. See https://pinia.vuejs.org/core-concepts/#Setup-Stores. We do not use options API except for PrimeVue component override.